kotlin-eclipse-core/src/org/jetbrains/kotlin/core/asJava/KotlinLightClassGeneration.kt

```diff
@@ -66,6 +67,11 @@ object KotlinLightClassGeneration {
                     }
                     
                     override fun shouldGenerateScript(script: KtScript): Boolean = false
+
+                    override fun shouldGenerateClassMembers(processingClassOrObject: KtClassOrObject): Boolean {
+						return shouldGenerateClass(processingClassOrObject) ||
+								processingClassOrObject.hasModifier(KtTokens.COMPANION_KEYWORD)
```

Comment:
Nice.
Minor, I'd suggest to use `super` to avoid leaky abstraction:
```
return super.shouldGenerateClassMembers(...) ||
    processingClassOrObject.hasModifier(KtTokens.COMPANION_KEYWORD)
```